tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -0,0 +1,434 @@
+        datasets["customer"],
+        destination="customer_profile",
+        event_timestamp_column="event_timestamp",
+        created_timestamp_column="created",
+    )
+    driver_ds = data_source_creator.create_data_source(
+        datasets["driver"],
+        destination="driver_hourly",
+        event_timestamp_column="event_timestamp",
+        created_timestamp_column="created",
+    )
+    orders_ds = data_source_creator.create_data_source(
+        datasets["orders"],
+        destination="orders",
+        event_timestamp_column="event_timestamp",
+        created_timestamp_column="created",
+    )
+    return {"customer": customer_ds, "driver": driver_ds, "orders": orders_ds}
+
+
+def construct_universal_feature_views(
+    data_sources: Dict[str, DataSource]
+) -> Dict[str, FeatureView]:
+    return {
+        "customer": create_customer_daily_profile_feature_view(
+            data_sources["customer"]
+        ),
+        "driver": create_driver_hourly_stats_feature_view(data_sources["driver"]),
+    }
+
+
+def setup_entities(
+    environment: "Environment", entities_override: Optional[Dict[str, List[Any]]] = None
+) -> "Environment":
+    environment.entities = (
+        entities_override if entities_override else construct_universal_entities()
+    )
+    return environment
+
+
+def setup_datasets(
+    environment: "Environment",
+    datasets_override: Optional[Dict[str, pd.DataFrame]] = None,
+) -> "Environment":
+    environment.datasets = (
+        datasets_override
+        if datasets_override
+        else construct_universal_datasets(
+            environment.entities, environment.start_date, environment.end_date
+        )
+    )
+    return environment
+
+
+def setup_data_sources(
+    environment: "Environment",
+    data_sources_override: Optional[Dict[str, DataSource]] = None,
+) -> "Environment":
+    environment.datasources = (
+        data_sources_override
+        if data_sources_override
+        else construct_universal_data_sources(
+            environment.datasets, environment.data_source_creator
+        )
+    )
+    return environment
+
+
+def setup_feature_views(
+    environment: "Environment",
+    feature_views_override: Optional[Dict[str, FeatureView]] = None,
+) -> "Environment":
+    environment.feature_views = (
+        feature_views_override
+        if feature_views_override
+        else construct_universal_feature_views(environment.datasources)
+    )
+    return environment
+
+
+@dataclass
+class Environment:
+    name: str
+    test_repo_config: TestRepoConfig
+    feature_store: FeatureStore
+    data_source_creator: DataSourceCreator
+
+    entities: Dict[str, List[Any]] = field(default_factory=dict)
+    datasets: Dict[str, pd.DataFrame] = field(default_factory=dict)
+    datasources: Dict[str, DataSource] = field(default_factory=dict)
+    feature_views: Dict[str, FeatureView] = field(default_factory=list)
+
+    end_date: datetime = field(
```

Comment:
Would it make more sense to move the default configuration into `test_repo_config`?